custom_components/atrea/climate.py

```diff
@@ -287,6 +326,7 @@ def set_fan_mode(self, fan_percent):
         if(fan_percent > 100):
             fan_percent = 100
         if(fan_percent >= 12 and fan_percent <= 100):
+            self.atrea.setProgram(AtreaProgram.TEMPORARY)
```

Comment:
Why are we changing program here to temporary? 

Changing of the fan speed should not change other modes. Please remove this line.

Reply:
Hmm, this is an interesting one. Just setting the fan mode without changing the program has no effect on my unit - the program remains set to AUTOMATIC, which means that the actual fan value remains set by the unit (and the desired fan speed is disregarded). This is also consistent with how `set_hvac_mode()` and `set_preset_mode()` are implemented - they change program, too. But I understand your point of view that `set_fan_speed()` should have no side effects. I see two ways out of this - either keeping it as is and have `set_fan_speed()` and `set_preset_mode()` both change the program to TEMPORARY, or introducing an explicit method for `set_program()` and removing the logic for setting the program from `set_fan_speed()`, `set_preset_mode()` and `set_hvac_mode()` and thus making those methods more transparent, but also relying on users having to explicitly call all required methods.

I'm fine with either approach, what's your preference?

Reply:
I see, maybe my unit works like that too - I never use automatic and my "Automatic" is achieved by hass automations and this addon.

What about changing to temporary only if current mode is automatic and in other cases leaving it like is? And then user can make an automation if they wish to turn it on via changing fan mode.

Reply:
Yep, that sounds like a good compromise, change pushed.